### PR TITLE
Add spec for m.room.pinned_events

### PR DESCRIPTION
**Pull request by turt2live**
I don't actually know if this is correct, as I can't build matrix-doc. Please let me know if I've done it wrong, and how to fix it.

Addresses https://github.com/matrix-org/matrix-doc/issues/550

### CODE REVIEW

**Comment by ara4n** [comment]
the spec looks fine to me (modulo the https://github.com/vector-im/riot-web/pull/5142#issuecomment-336670536 comment that in future we might want to consolidate pinning with general event tagging - e.g. upvoting/downvoting).  @richvdh, you are probably best positioned to say if it for spec/syntax validity?

**Comment by richvdh** [comment]
@matrixbot: test this please

File: .gitignore [hunk 1]
```diff
@@ -8,3 +8,4 @@
 /templating/out
 *.pyc
 *.swp
+/env
```
Comment (richvdh) [inline .gitignore]: nit: these are sorted at the moment. Could you stick it in the right place?

**Comment by turt2live** [comment]
@richvdh updated. I have no way of testing this and am going off of what code in the area looks like, so please pardon my mistakes if they're present.

**Comment by richvdh** [comment]
This is brilliant, thank you!

(writing some notes on how to build the spec locally is on my todo list...)

**Pull request merged**